lib/sessionmanager.js

```diff
@@ -22,14 +33,13 @@ SessionManager.prototype.logIn = function(req, user, options, cb) {
   
   var self = this;
   var prevSession = req.session;
-  
+
   // regenerate the session, which is good practice to help
   // guard against forms of session fixation
-  req.session.regenerate(function(err) {
+  this._delegate.regenerate(req, function(err) {
```

Comment:
You should change this to `self` here - I still experiencing issues with the `regenerate` and `save` methods identifying as undefined when using this implementation.

Reply:
I couldn't get it working with either `self` or `this`

Reply:
@imartinezmorales-loom this code worked for me with `cookie-parser` `0.6.0`. It works both with `self` or `this`

Reply:
I made the change. thanks for the review.